numpy/core/tests/test_multiarray.py

```diff
@@ -864,8 +864,12 @@ def test_casting(self):
 
         # Check that 'equiv' casting can reorder fields and change byte
         # order
+        # New in 1.12: This behavior changes in 1.13, test for dep warning
         assert_(np.can_cast(a.dtype, b.dtype, casting='equiv'))
-        c = a.astype(b.dtype, casting='equiv')
+        with warnings.catch_warnings(record=True) as log:
```

Comment:
Just to show it off (I really don't care about it), this could be an alternative:
```
with suppress_warnings() as sup:
    sup.filter(FutureWarning)
```
without checking that there are only FutureWarnings there, or if you want to make sure there is one future warning:
```
with suppress_warnings() as sup:
    l = sup.record(FutureWarning)
    ...
    assert_(len(l) > 0)
```
also works.